src/pymoca/backends/casadi/model.py

```diff
@@ -373,10 +373,27 @@ def simplify(self, options):
                     # For delayed states we do not have any reliable shape
                     # information available due to it being an arbitrary
                     # expression, so we just always expand.
-                    if (old_var.symbol._modelica_shape != (1,) or old_var.symbol.name() in self.delay_states):
+                    if (old_var.symbol._modelica_shape != ((None,),) or old_var.symbol.name() in self.delay_states):
                         expanded_symbols = []
-                        for ind in np.ndindex(old_var.symbol._modelica_shape):
-                            component_symbol = ca.MX.sym('{}[{}]'.format(old_var.symbol.name(), ",".join(str(i+1) for i in ind)))
+
+                        if old_var.symbol.name() in self.delay_states:
```

Comment:
A comment explaining what you're doing here (readying the formatted string) would be helpful. I knew because you told me, but otherwise I would have probably been very confused.